skyd/skyd.go

```diff
@@ -98,11 +99,11 @@ func (c *client) Pin(skylink string) (skymodules.SiaPath, error) {
 func (c *client) PinnedSkylinks() (map[string]interface{}, error) {
 	skylinksCache.mu.Lock()
 	defer skylinksCache.mu.Unlock()
-	// Check whether the cache is still valid and return it if so.
-	if skylinksCache.expiration.After(time.Now().UTC()) {
```

Comment:
REQ: This check got dropped. So we are always rebuilding the cache every call regardless of the expiration.

Reply:
No, we are never rebuilding the cache here. The cache's expiration is irrelevant right now. I thought of removing it but didn't.

The way we operate right now is:
1. Rebuild the cache at the start of each scan.
2. Consider the cache valid until the end of the scan, regardless of its length.
3. Go to 1.

Based on the assumptions being made, this is a safe and simple implementation which has minimal risk of blowing up in our faces.